src/main/fc/init.c

```diff
@@ -968,7 +968,7 @@ void init(void)
 #if defined(USE_CMS) && defined(USE_MSP_DISPLAYPORT) && defined(USE_OSD_OVER_MSP_DISPLAYPORT)
         case OSD_DISPLAYPORT_DEVICE_MSP:
             osdDisplayPort = displayPortMspInit();
-            if (osdDisplayPort || device == OSD_DISPLAYPORT_DEVICE_MSP) {
+            if (osdDisplayPort) {
```

Comment:
Not sure why we would not want to honor it if the user has explicitly configured the OSD device to be displayport? If we don't want to do this then we should at least be explicit and reset this in the configuration, and not just ignore the user's setting without feedback.

Reply:
If the MSP DisplayPort device fails to register there's really no point in using that for the OSD and thus initializing/running the OSD task.

Reply:
Fair enough - then we should be removing this option from the enum / configuration, as there is no point to allow the user to select something, and then completely ignore it - this will just lead to confusion and issues being opened.